### PR TITLE
Nose by default

### PR DESCRIPTION
**Pull request by millerdev**
It's been a long road. I hope this is the end.

@snopoke cc @benrudolph

### CODE REVIEW

File: corehq/apps/ota/tests/app_aware_restore.py [hunk 1]
```diff
@@ -10,6 +10,10 @@
 from corehq.apps.userreports.tests import get_sample_report_config
 from corehq.apps.users.models import CommCareUser
 
+def setupModule():
```
Comment (dimagimon) [inline corehq/apps/ota/tests/app_aware_restore.py]: E302 expected 2 blank lines, found 1

File: corehq/tests/noseplugins/dividedwerun.py [hunk 2]
```diff
@@ -6,6 +6,7 @@
 from nose.case import Test, FunctionTestCase
 from nose.plugins import Plugin
 from nose.suite import ContextSuite
+from nose.util import transplant_func
```
Comment (dimagimon) [inline corehq/tests/noseplugins/dividedwerun.py]: F401 'transplant_func' imported but unused

**Comment by millerdev** [comment]
@snopoke: we have a lot of settings files, particularly for the different test setups, and that makes the test runner fragile and hard to debug. One pain point has been the fact that [`dockersettings` imports `dev_settings`](https://github.com/dimagi/commcare-hq/blob/master/docker/dockersettings.py#L1), and `dockersettings` is imported by [`localsettings_docker`](https://github.com/dimagi/commcare-hq/blob/master/docker/localsettings_docker.py) (and indirectly by [`localsettings_docker_sharded`](https://github.com/dimagi/commcare-hq/blob/master/docker/localsettings_docker_sharded.py)) used by travis. We might want to change [this line](https://github.com/dimagi/commcare-hq/blob/master/dev_settings.py#L20) as long as we have `dev_settings` imported by travis. Do you think it would make sense to remove `dev_settings` from the mix on travis?

**Comment by millerdev** [comment]
Also is `.travis/localsettings.py` needed anymore?

**Comment by snopoke** [comment]
👍very happy to try and clean it up. I think i did it that way to try and
reduce duplication but it does make things more complicated.

**Comment by proteusvacuum** [comment]
@millerdev you've probably seen this, but thought I'd raise it:
http://nose.readthedocs.io/en/latest/index.html#note-to-users
> Nose has been in maintenance mode for the past several years and will likely cease without a new person/team to take over maintainership. New projects should consider using Nose2, py.test, or just plain unittest/unittest2.

`django-nose` documentation points to that project, and not `nose2`. Not sure if this is necessarily an issue though, so you can ignore me if you think it isn't important.
Otherwise, I'm very pumped for this change!

**Comment by millerdev** [comment]
@proteusvacuum yes, I saw that. It's unfortunate. That notice was posted after I started the effort of switching us over to nose. I briefly tried to use py.test instead, but it didn't look easy so I decided to continue on with nose until I have more time to investigate the possibility of switching to py.test. I'm leaning toward py.test because it is more mature than nose2.

**Comment by snopoke** [comment]
Test failing:
```
======================================================================
FAIL: testSubmitDuplicate (corehq.apps.receiverwrapper.tests.test_submit_errors.SubmissionErrorTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/mnt/corehq/util/test_utils.py", line 230, in inner
    return helper(*args, **kwargs)
  File "/mnt/corehq/util/test_utils.py", line 203, in __call__
    call_with_settings(fn_with_pre_and_post, run_config.settings, args, kwargs)
  File "/mnt/corehq/util/test_utils.py", line 218, in call_with_settings
    fn(*args, **kwargs)
  File "/mnt/corehq/util/test_utils.py", line 199, in fn_with_pre_and_post
    self.fn(*args, **kwargs)
  File "/mnt/corehq/apps/receiverwrapper/tests/test_submit_errors.py", line 56, in testSubmitDuplicate
    self.assertIn("Form is a duplicate", res.content)
AssertionError: 'Form is a duplicate' not found in '<OpenRosaResponse xmlns="http://openrosa.org/http/response"><message nature="submit_success">   \xe2\x88\x9a   </message></OpenRosaResponse>'
```

Looking into it. So far I've found that it works on master and only fails on this branch when run with other test classes (if you run `SubmissionErrorTest` class by itself it passes). What's odd is that I would n't expect this failure mode if it was a result of another test not cleanup up after itself.

**Comment by snopoke** [comment]
This seemed to fix the build: https://github.com/dimagi/commcare-hq/pull/11676/commits/d8e059770f389cb396705d4cc68b484d6647e4c4 but still not sure why

**Comment by snopoke** [comment]
:white_check_mark: :tada:

**Comment by millerdev** [comment]
✅ ✖️ 4️⃣2️⃣0️⃣8️⃣ 🐍

**Comment by benrudolph** [comment]
😂 😂 😂 😂 😂 
well done!!

**Comment by snopoke** [comment]
somebody should merge this

**Comment by benrudolph** [comment]
💣 💣

**Comment by snopoke** [comment]
:bangbang:

**Comment by proteusvacuum** [comment]
:tada: Amazing!

**Comment by millerdev** [comment]
@dimagi/team-commcare-hq: the default test runner is now nose. See [README.txt](https://github.com/dimagi/commcare-hq/blob/master/README.md#running-tests) for new details on how to run a subset of tests. Also, if you haven't already, check out [REUSE_DB](https://github.com/dimagi/commcare-hq/blob/master/corehq/tests/nose.py#L270-L286).

**Comment by nickpell** [comment]
💃

**Comment by kaapstorm** [comment]
:dancers::dancers::dancers:

**Comment by dannyroberts** [comment]
Woohoo!

**Comment by czue** [comment]
:nose: :tada:

File: dev_settings.py [hunk 3]
```diff
@@ -10,11 +10,11 @@
...
     'kombu.transport.django',
-    # for tests
-    'testapps.test_elasticsearch',
-    'testapps.test_pillowtop',
 )
 
+# TEST_RUNNER is overridden in testsettings, which is the default settings
+# module for the test command (see manage.py); this has no effect by default.
+# Use ./manage.py test --settings=settings to use this setting.
 TEST_RUNNER = 'testrunner.DevTestRunner'
```
Comment (snopoke) [inline dev_settings.py]: Is `.travis/localsettings.py` necessary anymore?

Comment (millerdev) [inline dev_settings.py]: Very good question. I'm working on a [branch to remove old/unused test code](https://github.com/dimagi/commcare-hq/compare/cleanup-tests), and I was not able to immediately able to determine if it's still needed. Maybe it's still used for JS tests?

Comment (snopoke) [inline dev_settings.py]: True, I think it is still used for the JS tests since they need a running server

**Pull request merged**